packages/babel-plugin-transform-classes/src/transformClass.js

```diff
@@ -349,20 +346,9 @@ export default function transformClass(
       call = t.logicalExpression("||", bareSuperNode, t.thisExpression());
     } else {
       bareSuperNode = optimiseCall(
```

Comment:
Looks like the `helper-optimise-call-expression` needs to be updated to support the `super(...args)` case; right now the `optimizeCall` is essentially a noop (or, well, it writes a `.call`, but not an "optimized" one)